### PR TITLE
cmake: partition_manager: define pm.yml for settings subsystem

### PR DESCRIPTION
**Pull request by hakonfam**
This pm.yml defines a partition 'settings_storage'.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
LGTM - need to clarify/fix CI failure (the file is in the tree...)

File: subsys/settings/pm.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#include <autoconf.h>
```
[the remark this answers is not in the document]
Comment (hakonfam) [reply subsys/settings/pm.yml]: As far as I can tell, it should be shared. Then it is up to the settings module to support multiple users, which I guess it does. But as you say, we should check that with someone more experienced with the settings module.

File: subsys/settings/pm.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+storage:
```
[the remark this answers is not in the document]
Comment (hakonfam) [reply subsys/settings/pm.yml]: I guess so, there was an issue with other components using this partition, but then I guess those should define their own.

[the remark this answers is not in the document]
Comment (hakonfam) [reply subsys/settings/pm.yml]: Fixed

File: subsys/fs/pm.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#include <autoconf.h>
+
+#ifdef CONFIG_FILE_SYSTEM_NFFS
```
Comment (oyvindronningstad) [inline subsys/fs/pm.yml]: Is this needed? I would guess it's covered by the corresponding `if` in the CmakeLists.txt.

Comment (hakonfam) [reply subsys/fs/pm.yml]: true, but it shows how to handle multiple subsystems defined in `fs`. That way each of them can configure in their own partition.

Comment (oyvindronningstad) [reply subsys/fs/pm.yml]: In that case, should the other be ` if (CONFIG_FILE_SYSTEM_NFFS)` or something?

Comment (hakonfam) [reply subsys/fs/pm.yml]: Not sure I get what you mean.  

One example could be
```
#ifdef CONFIG_FILE_SYSTEM_NFFS
nffs_storage:
  blabalbla
#endif

#ifdef CONFIG_FILE_SYSTEM_LITTLEFS
littlefs_storage:
  something: blabal
#endif
```

Does that answer your question?

Comment (oyvindronningstad) [reply subsys/fs/pm.yml]: Sorry, I meant `if (CONFIG_FILE_SYSTEM_FS)`

Comment (hakonfam) [reply subsys/fs/pm.yml]: Correct

File: subsys/fs/CMakeLists.txt [hunk 4]
```diff
@@ -16,6 +16,16 @@ if(CONFIG_FILE_SYSTEM)
   target_link_libraries_ifdef(CONFIG_FAT_FILESYSTEM_ELM   FS INTERFACE ELMFAT)
   target_link_libraries_ifdef(CONFIG_FILE_SYSTEM_LITTLEFS FS INTERFACE LITTLEFS)
   target_link_libraries_ifdef(CONFIG_FILE_SYSTEM_NFFS     FS INTERFACE NFFS)
+
+  if (CONFIG_FILE_SYSTEM_NFFS)
```
Comment (oyvindronningstad) [inline subsys/fs/CMakeLists.txt]: If multiple filesystems should be supported as stated in the comment below, I would say this `if` should be removed.

Comment (hakonfam) [reply subsys/fs/CMakeLists.txt]: agree

**Pull request merged**